service/gcs/main.go

```diff
@@ -20,6 +21,45 @@ import (
 	"go.opencensus.io/trace"
 )
 
+func kmsgLoop() {
+	f, err := os.Open("/dev/kmsg")
+	if err != nil {
+		logrus.WithError(err).Error("failed to open kmsg")
+		return
+	}
+	// /dev/kmsg never returns EOF, so we just sit in the loop waiting for more
+	// log entries forever.
+	for {
+		// Buffer size from CONSOLE_EXT_LOG_MAX	in Linux's printk.h.
+		b := [8192]byte{}
+		n, err := f.Read(b[:])
+		if err != nil {
+			logrus.WithError(err).Error("kmsg read failure")
+		} else {
+			line := string(b[:n])
+			ke, err := kmsg.Parse(line)
```

Comment:
is it guaranteed that we will only read one line at a time if we're using a buffer with 8192 bytes?

Reply:
Again, this is in the comment at the top.